llvm/lib/Target/SystemZ/SystemZISelLowering.cpp

```diff
@@ -920,6 +916,23 @@ bool SystemZTargetLowering::hasInlineStackProbe(const MachineFunction &MF) const
   return false;
 }
 
+// FIXME: Clang emits these casts always regardless of these hooks.
+TargetLowering::AtomicExpansionKind
+SystemZTargetLowering::shouldCastAtomicLoadInIR(LoadInst *LI) const {
```

Comment:
Looking at this again, I don't think this target hook should exist. Fundamentally AtomicExpandPass exists as a hack because SelectionDAG cannot handle emitting control flow for the cmpxchg blocks. Bitcasting types to match instructions is best done as part of regular legalization. cc @shiltian